test-apps/display-test-app/src/frontend/ViewAttributes.ts

```diff
@@ -545,7 +545,12 @@ export class ViewAttributes {
     const mapSettings = this.addMapSettings();
 
     const enableTerrain = (enable: boolean) => {
-      this.updateBackgroundMap({ applyTerrain: enable });
+      this.updateBackgroundMap({
+        // terrainSettings: {
+        //   providerName: "CuratedCesiumContent",
```

Comment:
Is this supposed to be commented out or not?

Reply:
this was commented out in [a166d5c](https://github.com/iTwin/itwinjs-core/pull/7436/commits/a166d5cf8cc595e0cd745361234fe032084f8eb6) to demonstrate the other alternative solution. i will drop that commit and clean things up, give me a few.

Reply:
Hmm this is a great catch. 

I don't think we can ever remove the current default cesium ion key terrain provider, as that is the only way for users to be able to pull in their own custom cesium assets.

I guess core could export a default providerName to use for this scenario that others should use to register and also use for when changing background map props via `vp.changeBackgroundMapProps` to use the specific provider.

Idk if there is a solution for providing fallbacks if apps register any custom properties in a saved view. Need to think about this more.

Reply:
That is only the best we can do if we accept the constraints that 1) we will not make a direct call to the service from core-frontend and 2) we won't provide a reusable package, instead requiring every app to reimplement it themselves and 3) we insist on requiring an iTwin ID to access data not associated with any iTwin. Even if we accept all those constraints, we can do at least somewhat better by permitting apps to *overwrite* the provider associated with the name "CesiumWorldTerrain" instead of introducing a new provider name.

To be clear: 99.9% of users do not care where the terrain comes from. They see it as data that **we** provide, all they care about is that when they open their saved view, they see the terrain. To my knowledge, exactly one user has created their own custom terrain provider. I think we should interpret "CesiumWorldTerrain" to mean "terrain supplied by iTwin platform, wherever it comes from".